consensus/src/lib.rs

```diff
@@ -635,18 +625,16 @@ impl<'a> ConsensusRef<'a> {
     }
 
     fn is_block_in_main_chain(&self, block_index: &BlockIndex) -> bool {
-        block_index.next_block_id.is_some()
+        block_index.get_next_block_id().is_some()
             || match self.db_tx.get_best_block_id().ok().flatten() {
                 Some(ref block_id) => block_index.get_block_id() == block_id,
                 None => false,
             }
     }
 
     /// Mark new block as an orphan
-    /// Ok(()) - Added
-    /// Err(BlockError) - StorageFailure  
     fn new_orphan_block(&mut self, block: Block) -> Result<(), BlockError> {
-        if block.get_prev_block_id().is_none() {
+        if block.get_prev_block_id().is_none() && block.is_genesis(self.chain_config) {
```

Comment:
I don't understand this condition.
Won't the condition `block.get_prev_block_id().is_none()` always be true for the genesis block?
If so, why not just check `block.is_genesis(..)`?

Reply:
Moved to another comment since this line is outdated. Resolving this.